Content.Client/Chameleon/ChameleonSystem.cs

```diff
@@ -62,12 +63,12 @@ private void OnShaderRender(EntityUid uid, SharedChameleonComponent component, B
         // unchanged.
         var parentXform = Transform(Transform(uid).ParentUid);
         var reference = args.Viewport.WorldToLocal(parentXform.WorldPosition);
-
+        var visibility = Getvisibility(uid, component);
         _shader.SetParameter("reference", reference);
-        _shader.SetParameter("stealthLevel", component.StealthLevel);
-        args.Sprite.Color = new Color(component.StealthLevel, component.StealthLevel, 1, 1);
+        _shader.SetParameter("visibility", visibility);
 
-        Dirty(component);
+        visibility = MathF.Max(0, visibility);
+        args.Sprite.Color = new Color(visibility, visibility, 1, 1);
```

Comment:
Another mistake of mine. I'm surprised colour didn't error when giving it negative values.